### PR TITLE
move exosphere data path

### PR DESCRIPTION
**Pull request by charlierudolph**
There goes all the home dir.

### CODE REVIEW

File: src/docker/tools/index.go [hunk 1]
```diff
@@ -61,8 +61,8 @@ func GetExitCode(containerName string) (int, error) {
 }
 
 // GetRenderedVolumes returns the rendered paths to the given volumes
-func GetRenderedVolumes(volumes []string, appName string, role string, homeDir string) ([]string, error) {
-	dataPath := path.Join(homeDir, ".exosphere", appName, role, "data")
+func GetRenderedVolumes(volumes []string, appName string, role string) ([]string, error) {
+	dataPath := path.Join("${APP_PATH}", ".exosphere", "data", role)
```
Comment (charlierudolph) [inline src/docker/tools/index.go]: This is the primary change.

File: src/docker/tools/index.go [hunk 2]
```diff
@@ -61,12 +61,9 @@ func GetExitCode(containerName string) (int, error) {
...
 	renderedVolumes := []string{}
-	if err := util.MakeDirectory(dataPath); err != nil { //nolint gas
```
Comment (hugoesthere) [inline src/docker/tools/index.go]: We'll still have to create the dir at whatever `${APP_PATH}/.exosphere/data/role` compiles to right?

**Review by hugoesthere — COMMENTED** [review]
We still have to set `APP_PATH` when running `docker-compose`: https://docs.docker.com/compose/compose-file/#variable-substitution

We can either do that for the user (i.e. in `docker/compose/index.go` commands) or we require the user to set it themselves. In which case we should probably name it something less generic than `APP_PATH`

File: src/docker/tools/index.go (hunk 2 above)
Comment (charlierudolph) [reply src/docker/tools/index.go]: True, I'll add that back

**Comment by charlierudolph** [comment]
`APP_PATH` is already set. Added in https://github.com/Originate/exosphere/pull/750

File: src/docker/tools/index.go [hunk 3]
```diff
@@ -61,14 +61,16 @@ func GetExitCode(containerName string) (int, error) {
 }
 
 // GetRenderedVolumes returns the rendered paths to the given volumes
-func GetRenderedVolumes(volumes []string, appName string, role string, homeDir string) ([]string, error) {
-	dataPath := path.Join(homeDir, ".exosphere", appName, role, "data")
+func GetRenderedVolumes(volumes []string, appDir string, role string) ([]string, error) {
+	appRelativePath := path.Join(".exosphere", "data", role)
+	realPath := path.Join(appDir, appRelativePath)
+	renderedPath := path.Join("${APP_PATH}", appRelativePath)
```
Comment (hugoesthere) [inline src/docker/tools/index.go]: If `appDir` is already something like `Users/charlie/exosphere/space-tweet`, and `${APP_PATH}` is also `Users/charlie/exosphere/space-tweet`, won't this line combine them into `Users/charlie/exosphere/space-tweet/Users/charlie/exosphere/space-tweet`? That seemed to be the behavior when I tested it out here: https://golang.org/pkg/path/#Join

File: test_helpers/index.go [hunk 4]
```diff
@@ -73,25 +73,22 @@ func createEmptyApp(appName, cwd string) (string, error) {
...
 	if err != nil {
-		return errors.Wrap(err, "Failed to kill test containers")
+		return err
 	}
-	testComposeProjectName := composebuilder.GetTestDockerComposeProjectName(appName)
-	err = application.CleanServiceTestContainers(appDir, testComposeProjectName, writer)
-	if err != nil {
-		return errors.Wrap(err, "Failed to kill test containers")
+	if doesExist {
+		cmdPlus := execplus.NewCmdPlus("exo", "clean") // nolint gas
```
Comment (hugoesthere) [inline test_helpers/index.go]: `killTestContainers` was used for creating dangling images/volumes, I think it was more specific than `exo clean`. Are the tests still doing that?

File: src/docker/tools/index.go (hunk 3 above)
Comment (charlierudolph) [reply src/docker/tools/index.go]: This is joining `${APP_PATH}` with `path.Join(".exosphere", "data", role)`.

Comment (charlierudolph) [reply src/docker/tools/index.go]: I'll rename `appRelativePath` to `dataRelativePath`

File: test_helpers/index.go (hunk 4 above)
Comment (charlierudolph) [reply test_helpers/index.go]: Auhh. I will revert the changes here and add a new function to clean up at the end of a scenario (which was the other place using this)

**Comment by charlierudolph** [comment]
@hugobho comments addressed

**Review by hugoesthere — APPROVED** [review]

**Pull request merged**